### PR TITLE
Fix ExplicitInitRule correction

### LINKED ISSUE #5289: Autocorrect bug with Explicit Init rule
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Steps to reproduce

```shell
$ swiftlint --version
0.53.0

$ echo "f { e in\n    A.init(e: e)\n}" >> z1.swift

$ cat z1.swift
f { e in
    A.init(e: e)
}

$ swiftlint z1.swift --autocorrect --enable-all-rules
Correcting Swift files at paths z1.swift
Correcting 'z1.swift' (1/1)
/Users/.../Documents/z1.swift:2:6 Corrected Explicit Init
Done correcting 1 file!

$ cat z1.swift
f { e inA(e: e)
}
```

The lint result should be

```swift
f { e in
    A(e: e)
}
```

### PR DESCRIPTION
**Pull request by BB9z**
fix #5289 

The CI is having issues running now, you can verify it locally.

### CODE REVIEW

File: Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift [hunk 1]
```diff
@@ -13,10 +13,24 @@ class ExplicitInitRuleTests: SwiftLintTestCase {
...
+                A↓.init(e: e)
+            }
+            """):
+                Example("""
+                f { e in
+                    A(e: e)
+                }
+                """)
+        ]
+        correction.merge(ExplicitInitRule.description.corrections) { current, _ in current }
```
Comment (SimplyDanny) [inline Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift:28]: The example can be added directly to `ExplicitInitRule.description.corrections`. No need to update it here.

File: Source/SwiftLintBuiltInRules/Rules/Idiomatic/ExplicitInitRule.swift [hunk 2]
```diff
@@ -201,7 +201,7 @@ private extension ExplicitInitRule {
             }
 
             correctionPositions.append(violationPosition)
-            let newNode = node.with(\.calledExpression, calledBase.trimmed)
+            let newNode = node.with(\.calledExpression, calledBase.with(\.trailingTrivia, []))
```
Comment (SimplyDanny) [inline Source/SwiftLintBuiltInRules/Rules/Idiomatic/ExplicitInitRule.swift:204]: Can also be just that:
```suggestion
            let newNode = node.with(\.calledExpression, calledBase)
```
But yours is fine as well. 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -22,6 +22,10 @@
 
 #### Bug Fixes
 
+* Fix ExplicitInitRule correction.
```
Comment (SimplyDanny) [inline CHANGELOG.md:25]:
```suggestion
* Fix correction of `explicit_init` rule by keeping significant trivia.  
```

File: Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift [hunk 4]
```diff
@@ -13,10 +13,24 @@ class ExplicitInitRuleTests: SwiftLintTestCase {
             Example("foo(↓.init())")
         ]
 
+        var correction = [
+            Example("""
+            f { e in
+                A↓.init(e: e)
+            }
```
Comment (SimplyDanny) [inline Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift:20]: Could you also add a comment to show that it is neither removed:
```suggestion
            f { e in
                // comment
                A↓.init(e: e)
            }
```

**Review by SimplyDanny — COMMENTED** [review]
Thanks! Looks good in general.

File: Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift (hunk 4 above)
Comment (BB9z) [reply Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift]: I think there is no need to add special comments here. It's fine as long as future tests go well.

Comment (SimplyDanny) [reply Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift]: The comment is not supposed to serve documentation purposes but to ensure that comments don't get lost either when the rule rewrites code.

Comment (BB9z) [reply Tests/SwiftLintFrameworkTests/ExplicitInitRuleTests.swift]: Oh, understood what you mean now. Added it.

**Review by SimplyDanny — APPROVED** [review]

**Pull request merged**